### DISCUSSION TITLE
5 Fans, 1 AIO & GPU - Only 3 fans have control.

### DISCUSSION THREAD

**Discussion by ZomBenTV** [Rem0o/FanControl.Releases#755] [Q&A]
Hey, I just downloaded this app yesterday and I see so much potential. I even love it as it stands with only partial functionality where I cannot use the program to it's fullest.

When I launch the program it shows 8 or 9 devices in total

The GPU
6 Devices showing RPM
2 Devices not showing any RPM

I can control the GPU fine
I can control 3 of the 5 fans but the other 2 I can't identify because while I can see the RPM I can't stop the fan to identify
I cannot do anything with the other device which I assume is the AIO but I again cannot control it so I can't tell which it is.

Can anyone give me advice on how to fix this issue?

Thanks!

**Comment by xhul-dev** [comment]
Hello there.
Did you run the assisted setup?

**Reply by ZomBenTV** [reply]
Yes I did

**Comment by xhul-dev** [comment]
Ok, so we'll have to use our brain.
I must confess, i'm having some trouble identifying how many physical fans you really have, and how they are actually connected to the motherboard, but that's probably my understanding skills that suck.

As a first step, could you please clarify what the physical environement is (actual, not what Fan Control says):
- case fans (non-AIO): 5? Are they using one motherboard connector each?
- AIO fans: How many? How are they connected to the motherboard?
- GPU fans: How many?

As a second step, can you upload a screenshot of what's to be seen in Fan Control?
Ideally, right after Fan Control was executed for the first time (assisted setup included).

Thanks in advance.

**Reply by ZomBenTV** [reply]
![Fan Controller](https://user-images.githubusercontent.com/77062639/167627623-f7b66916-c53b-477a-a9a8-cac820a0020e.JPG)

So, I have 5 fans in total. One on the Rear, 2 on the top (only the one closest to the rear is controllable) and 2 on the front radiator (only the one on the top is controllable) The GPU has 2 fans and both are controllable.

Thankfully all fans spin, it's just that I can't control and identify all of them.

All of them are connected directly to the mobo.

Thanks for the help btw.

**Reply by ZomBenTV** [reply]
When I try the ones labeled as #1 ???, #2 ??? & #3 ???
They all stay at max speed. I've tried using the fan identifier, Manual, Curve, GPU, Cpu, Mix and Flat modes and they just won't change.

**Reply by Rem0o** [reply]
In your motherboard BIOS, are these headers set to PWM while the others are DC or vice-versa? Do they have a "smart curve" or whatever mode enabled on them?

**Reply by ZomBenTV** [reply]
I'll have to check that tomorrow. I'll boot up the BIOS and confirm that. I'm getting ready for work at the moment.

I'll take some shots of the BIOS with my phone as I'm not 100% sure about the header types. I'll grab the Mobo manual and see.

FYI the Mobo is a Gigabyte AORUS Z370 Gaming 7

**Reply by ZomBenTV** [reply]
I just remembered having taken this shot yesterday on my phone. Does this help?

![20220509_110048](https://user-images.githubusercontent.com/77062639/167681668-df5870f2-03da-4574-832d-4cf9666be46f.jpg)

**Reply by ZomBenTV** [reply]
I'm guessing by this pic I might just have some cables put in the wrong spots?

I can only assume, without doing any disconnects my AIO should be plugged into the CPU fan andy other 5 fans in the System 1-5 because I'm not running a standalone waterloo/pump.

**Reply by ZomBenTV** [reply]
I'm doing lots of research while I'm at work. It's dead here.

My Mobo: Gigabyte AORUS Z370 Gaming 7 has the following fan headers.

1 x CPU fan header
1 x Water cooling CPU fan header
4 x System fan Headers
1 x 3 Amp fan/water cooling pump header
1 x System fan/water cooling header

So based off that I can only assume that the best approach for connecting my fans would be to set them like this

Plug the AIO in the CPU fan header
Plug 4 of my fans in the regular System fan Headers 1-4

But which is ideal for the 5th fan?

**Reply by ZomBenTV** [reply]
> 

I didn't have much time to mess around with them today but they were all on Auto, I set them to PWM to give it a try. No change noticed.

As for smart curve it looks like they have something set but I honestly forgot to take notice.

Thursday I should have actual free time to dive deep into the trouble shooting. Hopefully I can get this all resolved.

**Comment by xhul-dev** [comment]
So, 7 physical fans total, thanks.
That makes a bit more sense now, especially since amongst the 9 that FC detects, 2 of them are at 0 RPM (maybe the pump is detected as a fan, no idea for the second one).

You stated that the GPU has 2 controllable fans, yet i only see 1 labelled for the GPU, maybe the second GPU fan is one of the 3 unknown?

Also, for the top and front fans issue, are you sure those are separately connected to the MB? (by that i mean, not using a 2>1 physical adapter)
If yes, I would check how they are parametered in the BIOS, just as Rem0o suggested.

**Reply by ZomBenTV** [reply]
When I test the GPU fans they both are controlled as one. So if I go to the GPU selected, change it to manual and set it to zero, they both turn off.

Also, I have all fans connected directly to the Mobo with no splitters or hubs.

I'm assuming that this program would work best with direct Mobo connections or are fan hubs a viable/better option?

**Reply by xhul-dev** [reply]
> When I test the GPU fans they both are controlled as one. So if I go to the GPU selected, change it to manual and set it to zero, they both turn off.

That's a possibility indeed, just how can you know if are they both affected if you only have 1 GPU card in FC?
Maybe you checked the rotation physically?

> Also, I have all fans connected directly to the Mobo with no splitters or hubs.
> 
> I'm assuming that this program would work best with direct Mobo connections or are fan hubs a viable/better option?

Yeah, you'll get more control without hubs or splitters.

To identify the unknown fans, here is how i would do it:
1) Make sure all controls are turned off in FC (left-right toggle button), to only have the BIOS in control.
2) Shutdown.
3) Disconnect all fans physically except top-front.
4) Check FC speed cards, amongst 1|2|3, the one that shows a non-null RPM is the one.

Repeat the operation with front-bottom.
Note that if the second GPU fan is amongst 1|2|3 as i suspect, FC could show 2 fans with a non-null RPM, not 1.

**Reply by ZomBenTV** [reply]
I haven't had the time to disconnect/reconnect fans yet. Thursday I'm off and I'll have loads of time to muck about with it all.

In the meantime, I might have alot of valuable info attached to the previous response where Rem0o responded.

**Comment by ZomBenTV** [comment]
@xhul-dev & @Rem0o

I've done some more tinkering today. I ended up staying home from work. I checked my fan headers, moved them around and redid the auto set up.

I also went to the BIOS & checked for the the status of Auto/DC/PWM and switched it all to PWM.

This is how everything is looking now. Not a whole lot different but it also seems like nothing is running at max speed anymore. But there is still 2 fans I cannot control.
![20220511_183553](https://user-images.githubusercontent.com/77062639/167956398-4b93bb73-41d4-406d-a933-e9d48b5241d0.jpg)
![20220511_183614](https://user-images.githubusercontent.com/77062639/167956400-c15d7666-6b69-4275-9850-5610024ac6f4.jpg)
![20220511_185939](https://user-images.githubusercontent.com/77062639/167956401-50668703-b4a2-4e9d-861d-ec61bd302e6d.jpg)
![20220511_190100](https://user-images.githubusercontent.com/77062639/167956404-59a21825-d087-4955-9e02-4a62c09921f7.jpg)

I hope this can help solve the problem.

I'll do a fresh screen grab of the app when I boot up again in a little bit. I thought I had them on my phone but apparently I did not.

**Reply by xhul-dev** [reply]
> But there is still 2 fans I cannot control.

In the BIOS aswell?
If yes, which ones are those? (according to the listing on your screenshot)

**Comment by jmarucha** [comment]
Hey,
About AIO, I just made a plugin for FanControl to place nicely with liquidctl.
So far I tested it on my NZXT Kraken X63 and can do pump, and get info from coolant temp sensors.
If you wanna test my plugin (may help you, for my case the control of pump speed did work erratically via motherboard controller), please give it a go and let me know how it works
https://github.com/jmarucha/FanControl.Liquidctl
